spec/models/user_spec.rb

```diff
@@ -1 +1,74 @@
 require 'rails_helper'
+RSpec.describe User, type: :model do
+  describe 'Valid?' do
+    let(:user) { build(:user) }
+    it 'valid with valid attributes' do
+      expect(user).to be_valid
+    end
+  end
+
+  describe '.filter' do
+    let(:first_project) { create(:project, name: 'first_project') }
+    let(:second_project) { create(:project, name: 'second_project') }
+
+    let(:user_foo) { create(:user, first_name: 'user_foo', project: first_project) }
+    let(:user_bar) { create(:user, first_name: 'user_bar', project: first_project) }
+    let(:user_john) { create(:user, first_name: 'user_john', project: second_project) }
+
+    context 'call without param' do
+      it 'returns all users' do
+        expected_all_user_ids = [user_foo.id, user_bar.id, user_john.id]
+        expect(User.filter(nil).map(&:id)).to eq expected_all_user_ids
+      end
+    end
+
+    context 'when method with id Project.first' do
+      it 'returns users list by Project.first.id' do
+        expected_user_ids = [user_foo.id, user_bar.id]
+        expect(User.filter(first_project.id).map(&:id)).to eq(expected_user_ids)
+      end
+    end
+  end
+
+  describe '.as_json' do
+    context 'when array of instances converting to json' do
+      let(:project) { create(:project) }
+      let(:user) { create(:user, project: project) }
+
+      it 'returns an array of objects as json' do
+        as_json = JSON.parse(User.as_json([user])).first
+
+        user_attributes = %w[first_name last_name email phone]
+        project_attributes = %w[name desc]
+
+        user_attributes.each { |attr| expect(as_json[attr]).to eq(user[attr]) }
+        project_attributes.each { |attr| expect(as_json[attr]).to eq(user[attr]) }
+      end
+    end
+  end
+
+  describe '.from_omniauth' do
+    context 'when user use Google SSO' do
+      let(:access_token) { OmniAuth.config.mock_auth[:google] }
+      let(:registered_google_user) { create(:user, email: access_token[:info][:email]) }
+      let(:new_user) { build(:user, email: access_token[:info][:email]) }
+
+      it 'returns not to change User count' do
```

Comment:
I do not understand this test. What do you test in?

Reply:
@VasylYaremkiv 
When google user registered in the application, this test check if returned not changed count of users.